packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx

```diff
@@ -539,7 +539,43 @@ export class DropdownBase extends React.Component<IDropdownInternalProps, IDropd
   private _onRenderList = (props: ISelectableDroppableTextProps<IDropdown, HTMLDivElement>): JSX.Element => {
     const { onRenderItem = this._onRenderItem } = props;
 
-    return <>{props.options.map((item: any, index: number) => onRenderItem({ ...item, index }, this._onRenderItem))}</>;
+    const groups: IDropdownOption[][] = [[]];
+    let index = -1;
+
+    for (const item of props.options) {
+      const groupLength = groups.length;
+      const currentGroup = groups[groupLength - 1];
+      if (item.itemType === SelectableOptionMenuItemType.Header) {
+        if (currentGroup.length === 0) {
+          // if current group is empty i.e. header is first item
+          currentGroup.push(item);
+        } else {
+          // otherwise create a new group and push header to new group
+          groups[groupLength] = [];
+          groups[groupLength].push(item);
+        }
+      } else {
+        // push all non headers to the current group
+        currentGroup.push(item);
+      }
+    }
+
+    const renderedGroups = [];
+    for (const group of groups) {
+      const groupId = getId('group');
+      // check if first item in group is a header
+      const hasHeader = group[0].itemType && group[0].itemType === SelectableOptionMenuItemType.Header;
+      if (hasHeader) {
+        group[0].id = groupId;
+      }
+      const items = group.map((item: any) => {
+        index++;
+        return onRenderItem({ ...item, index }, this._onRenderItem);
+      });
+      renderedGroups.push(hasHeader ? <div aria-labelledby={groupId}>{items}</div> : items);
```

Comment:
><div aria-labelledby={groupId}> [](start = 38, length = 31)

consider adding `role="group"` to this div to help make sure screen readers give a consistent experience

Reply:
good catch. i'd added and removed it a few times in testing. seems i forgot to add it back